src/main/flight/pid.c

```diff
@@ -1343,7 +1351,13 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, timeUs_t currentTim
         // b = 1 and only c (feedforward weight) can be tuned (amount derivative on measurement or error).
 
         // -----calculate P component
-        pidData[axis].P = pidCoefficient[axis].Kp * errorRate * tpaFactorKp;
+        // If enabled vary P and D using the content of AUXn (Heli-like vehicles, gyro-sensitivity channel).
+        float auxPgain = 1;
+        int8_t auxCH = pidAdjust()->pid_adjust_aux_channel;
+        if (auxCH > 3) {
+            auxPgain = ((-0.1f * constrainf(((float)rcData[auxCH]), 1000.0f, 2000.0f)) + 200.0f) / 50.0f;
```

Comment:
This uses a lot of magic numbers. Using defines for them will make it easier to understand what is being done.

Reply:
scale_rangef will be quite self-explaining ...

Reply:
@mikeller @ledvinap 
The RC channel should be scaled to +/- 100% with zero at 1500. 
Can `scale_rangef` do this?

BTW the formulas to scale something are simple math. Also you can find it in engineer's handbook.

Reply:
If you need an engineer's handbook to figure out what a section of code does, then it is not well written...

Reply:
Will this work?
`auxPgain = (scaleRangef((float)rcData[pidAdjustmentChannel], 1000.0f, 2000.0f, 100.0f, 0.0f)) / 50.0f;`

Reply:
Kind of ... PWM range is PWM_RANGE_MIN .. PWM_RANGE_MAX, not 1000...2000
Also there is not need to post-scale value - simply use 2f.0 as output range. And there is still problem with disabled pid at range maximum ...

Reply:
@ledvinap 
From my point of view using an output range different from 100 (per cent) to 0 would be as self -explaining as to magical numbers i used before, but this is just one of 1000s opinions, so no issue to me.

The way i did it, the result matches the data on the display of my TX. If it states +50%, the gyro sensitivity is reduced by 50%, if it states -50% the gyro sensitivity is increased by 50%.

Correct, i will have to use `PWM_RANGE_MIN/MAX`.